dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/subworkflow/SubWorkflowAsyncTaskExecuteFunction.java

```diff
@@ -55,6 +55,8 @@ public SubWorkflowAsyncTaskExecuteFunction(TaskExecutionContext taskExecutionCon
         }
         subWorkflowInstance = processInstanceDao.queryById(subWorkflowInstance.getId());
         if (subWorkflowInstance != null && subWorkflowInstance.getState().isFinished()) {
+            // update the varPool of sub workflow
+            taskExecutionContext.setVarPool(subWorkflowInstance.getVarPool());
```

Comment:
This block is a method to obtain status. Why do you need to set variables here?

Reply:
Yes, we need to set the variable in the task instance, since we didn't have a good way in the engine, this should be refactor after `Master refactor`.